http-client/src/client.rs

```diff
@@ -79,4 +78,62 @@ impl Client for HttpClient {
 		}?;
 		jsonrpc::from_value(json_value).map_err(Error::ParseError)
 	}
+
+	async fn batch_request<T, M, P>(&self, batch: Vec<(M, P)>) -> Result<Vec<T>, Error>
+	where
+		T: DeserializeOwned + Default + Clone,
+		M: Into<String> + Send,
+		P: Into<jsonrpc::Params> + Send,
+	{
+		let mut calls = Vec::with_capacity(batch.len());
+		// NOTE(niklasad1): `ID` is not necessarily monotonically increasing.
+		let mut ordered_requests = Vec::with_capacity(batch.len());
```

Comment:
I realized that we could have an additional request_set `HashSet<RequestID>`  here to do quick lookups to avoid O(n) lookups